src/editor/isometric_editor_plugin.cpp

```diff
@@ -361,6 +365,22 @@ void IsometricEditorPlugin::_on_plane_visibility_timeout(int p_plane_type) {
     map_handling_data.plane_timers[p_plane_type] = Ref<SceneTreeTimer>();
 }
 
+void IsometricEditorPlugin::_on_map_size_changed() {
+    auto index{reinterpret_cast<uint64_t>(selected_map)};
+    MapHandlingData& map_handling_data{handling_data_map[index]};
+    const Vector3& map_size{selected_map->get_size()};
+
+    map_handling_data.editor_planes[EditorPlane::PlaneType::EDITOR_DRAWER].set_position(0);
+    map_handling_data.editor_planes[EditorPlane::PlaneType::X_MIN_VIEW_LIMITER].set_position(0);
+    map_handling_data.editor_planes[EditorPlane::PlaneType::X_MAX_VIEW_LIMITER].set_position(map_size.x);
```

Comment:
The limits are reset to the new boundaries of the map, but when possible they should stay in the same position, and potentially clamped if the former position is no longer available.